content/platforms/kubernetes/tasks/ldap-on-k8s.md

```diff
@@ -0,0 +1,501 @@
+```
+
+### Deploying a Redis Enterprise Cluster
+A Redis Enterprise cluster requires no special setup other to be configured with LDAP as the configuration is outside of the scope of the Redis Enterprise CR.
+
+You can create a simple test cluster that can be used with the examples below via this CR:
+
+```
+apiVersion: app.redislabs.com/v1
+kind: RedisEnterpriseCluster
+metadata:
+  name: ldap
+spec:
+  nodes: 3
+  redisEnterpriseNodeResources:
+    limits:
+      cpu: 1
+      memory: 3Gi
+    requests:
+      cpu: 1
+      memory: 3Gi
+```
+
+### LDAP Configuration
+
+A new Redis Enterprise cluster does not have any LDAP configuration but is ready
+to accept them. The saslauthd daemon is already running and configured to use
+LDAP as a mechanism. All you need to do is to provide the cluster
+with the LDAP configuration information (see [saslauthd’s configuration reference](https://github.com/cyrusimap/cyrus-sasl/blob/master/saslauthd/LDAP_SASLAUTHD)).
+
+Using the example LDAP server we setup, a minimal set of configuration parameters for saslauthd:
+
+```
+ldap_servers: ldap://ldap-openldap.bdb.svc:389
+ldap_search_base: ou=users,dc=example,dc=org
+ldap_filter: (uid=%u)
+ldap_bind_dn: cn=admin,dc=example,dc=org
+ldap_password: admin
+```
+
+### Using rladmin to update the configuration
+If this information was in a file called “ldap.conf”, you can connect to a Redis Enterprise Node pod and configure the server via:
+
+```
+rladmin cluster config saslauthd_ldap_conf ldap.conf
+```
+
+The saslauthd daemon is configured to use the file `/etc/opt/redislabs/saslauthd.conf` on each node. You cannot edit this file directly on the node. Instead, you just need a local copy of the configuration you desire and rladmin command will update the cluster configuration. The cluster will update all the node’s configuration and preserve the setting after any pod restarts.
+
+### Using the REST API to update the configuration
+Alternatively, you can post the same configuration data in a request to the REST API for the cluster. First you forward the cluster API from a pod:
+
+```
+kubectl port-forward pod/ldap-0 9443
+```
+
+Then you change the LDAP configuration with a JSON version of the ldap configuration.
+
+```
+cat << EOF > tmp.json
+{"saslauthd_ldap_conf": "ldap_servers: ldap://ldap-openldap.bdb.svc:389\nldap_search_base: ou=users,dc=example,dc=org\nldap_filter: (uid=%u)\nldap_bind_dn: cn=admin,dc=example,dc=org\nldap_password: admin\n"}
+EOF
+curl -f -k -u "demo@redislabs.com:xxx" -X PUT -d @tmp.json -H "Content-Type: application/json" https://localhost:9443/v1/cluster
+```
+
+The key value for “saslauthd_ldap_conf” is the configuration file verbatim as it would be in the saslauthd configuration file. As such, it can be relatively tricky to get the syntax right and generating this programmatically is helpful. A little python program (make_request.py) can help:
```

Comment:
Please ensure that the names of processes like saslauthd are between back ticks: `saslauthd`